distributed/diagnostics/worker_monitor.py

```diff
@@ -0,0 +1,35 @@
+from collections import defaultdict
+
+from ..utils import ignoring
+
+with ignoring(ImportError):
+    from bokeh.plotting import figure
+    from bokeh.models import ColumnDataSource, Range1d
+
+def resource_profile_plot(width=600, height=400):
+    names = ['times','workers', 'cpu', 'memory-percent']
+    source = ColumnDataSource({k: [] for k in names})
+
+    p = figure(width=width, height=height, x_axis_type='datetime')
+    p.multi_line(xs='times', ys='cpu', line_width=3, color='red', source=source)
+    p.multi_line(xs='times', ys='memory-percent', line_width=3, color='blue', source=source)
+
+    return source, p
+
+def resource_profile_update(source, worker_buffer, times_buffer):
+    data = defaultdict(list)
+
+    workers = sorted(worker_buffer[-1], reverse=True)
+
+    import pdb; pdb.set_trace()
+
+    for name in ['cpu', 'memory-percent']:
+        try:
+            data[name] = [[msg[w][name] for msg in worker_buffer if msg != {} and w in msg] for w in workers]
+        except KeyError:
+            pass
+
+    data['workers'] = workers
+    data['times'] = [[t for i, t in enumerate(times_buffer) if worker_buffer[i] != {}] for w in workers]
+
+    source.data.update(data)
```

Comment:
Can I ask for a test similar to what is in `diagnostics/tests/test_status_monitor.py`.  I also find that writing to a test first before interacting with bokeh makes me develop much more quickly.  The test can setup everything and try it far faster than I can by manually setting up scheduler, workers, etc..

To test a particular testing file I run the following command from within the repository's root directory.

    py.test distributed/diagnostics/tests/test_status_monitor.py

If you copy the model in that file will spin up a scheduler, workers, and executor for you and shut them down cleanly in around a second.